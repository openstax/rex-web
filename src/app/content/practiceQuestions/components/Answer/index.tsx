import { HTMLElement } from '@openstax/types/lib.dom';
import React from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { LinkedArchiveTreeSection } from '../../../types';
import { PracticeAnswer, PracticeQuestion } from '../../types';
import {
  AnswerAlignment,
  AnswerBlock,
  AnswerContent,
  AnswerExcerpt,
  AnswerIndicator,
  AnswerInput,
  AnswerWrapper,
  StyledAnswerResult,
} from './styled';

interface AnswerResultProps {
  showCorrect: boolean;
  isSubmitted: boolean;
  isSelected: boolean;
  isCorrect: boolean;
}

// tslint:disable-next-line: variable-name
const AnswerResult = ({ showCorrect, isSelected, isSubmitted, isCorrect }: AnswerResultProps) => {
  const resultMsgKey = isCorrect
    ? 'i18n:practice-questions:popup:correct'
    : 'i18n:practice-questions:popup:incorrect';

  if (
    !(isSubmitted && isSelected)
    && !(isSubmitted && showCorrect && isCorrect)
  ) {
    return null;
  }

  return <StyledAnswerResult isCorrect={isCorrect}>
    <FormattedMessage id={resultMsgKey}>
      {(msg) => msg}
    </FormattedMessage>
  </StyledAnswerResult>;
};

interface AnswerProps {
  isSelected: boolean;
  isSubmitted: boolean;
  showCorrect: boolean;
  answer: PracticeAnswer;
  question: PracticeQuestion;
  choiceIndicator: string;
  onSelect: () => void;
  source: LinkedArchiveTreeSection;
}

// tslint:disable-next-line: variable-name
const Answer = ({
  answer,
  question,
  showCorrect,
  choiceIndicator,
  isSubmitted,
  isSelected,
  onSelect,
}: AnswerProps) => {
  const answerRef = React.useRef<HTMLElement>(null);
  const isCorrect = answer.correctness === '1.0';

  React.useEffect(() => {
    if (showCorrect && isCorrect && answerRef.current) {
      answerRef.current.focus();
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [showCorrect, isCorrect]);

  return <AnswerWrapper tabIndex={-1} ref={answerRef}>
    <AnswerInput
      id={choiceIndicator}
      type='radio'
      name={question.uid}
      checked={isSelected}
      disabled={isSubmitted}
      onChange={onSelect}
    />
    <AnswerBlock
      tabIndex={-1}
      htmlFor={choiceIndicator}
      showCorrect={showCorrect}
      isCorrect={isCorrect}
      isSubmitted={isSubmitted}
      isSelected={isSelected}
    >
      <AnswerIndicator
        aria-label={useIntl().formatMessage(
          {id: 'i18n:practice-questions:popup:answers:choice'},
          {choiceIndicator: choiceIndicator.toUpperCase()}
        )}
      >{choiceIndicator}</AnswerIndicator>
      <AnswerAlignment>
        <AnswerContent>
          <AnswerExcerpt dangerouslySetInnerHTML={{ __html: answer.content_html }} />
          <AnswerResult
            showCorrect={showCorrect}
            isSelected={isSelected}
            isSubmitted={isSubmitted}
            isCorrect={isCorrect}
          />
        </AnswerContent>
      </AnswerAlignment>
    </AnswerBlock>
  </AnswerWrapper>;
};

export default Answer;
