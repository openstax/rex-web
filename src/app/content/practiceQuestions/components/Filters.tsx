import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components/macro';
import ChapterFilter from '../../components/popUp/ChapterFilter';
import Filters, { FilterDropdown, FiltersTopBar } from '../../components/popUp/Filters';
import { FiltersChange } from '../../components/popUp/types';
import { LinkedArchiveTreeSection } from '../../types';
import { setSelectedSection } from '../actions';
import * as selectors from '../selectors';

// tslint:disable-next-line: variable-name
const StyledChapterFilters = styled(ChapterFilter)`
  padding: 0;
`;

export default () => {
  const [open, setOpen] = React.useState(false);
  const locationFilters = useSelector(selectors.practiceQuestionsLocationFilters);
  const selectedSection = useSelector(selectors.selectedSection);
  const dispatch = useDispatch();
  const setFilters = React.useCallback((change: FiltersChange<LinkedArchiveTreeSection>) => {
    const clickedSection = change.new.pop();
    if (!clickedSection) {
      // user clicked on the selected section
      setOpen(false);
      return;
    }
    dispatch(setSelectedSection(clickedSection));
    setOpen(false);
  }, [dispatch]);
  const selectedLocationFilters = React.useMemo(
    () => new Set(selectedSection ? [selectedSection.id] : []),
    [selectedSection]);

  return <Filters>
    <FiltersTopBar>
      <FilterDropdown
        label='i18n:practice-questions:popup:filters:chapters'
        ariaLabelId='i18n:practice-questions:popup:filters:filter-by:aria-label'
        dataAnalyticsLabel='Filter PQ by Chapter & Section'
        open={open}
        setOpen={(isOpen: boolean) => setOpen(isOpen)}
      >
        <StyledChapterFilters
          locationFilters={locationFilters}
          selectedLocationFilters={selectedLocationFilters}
          setFilters={setFilters}
          multiselect={false}
          ariaLabelItemId='i18n:practice-questions:popup:filters:filter-by:aria-label'
        />
      </FilterDropdown>
    </FiltersTopBar>
  </Filters>;
};
