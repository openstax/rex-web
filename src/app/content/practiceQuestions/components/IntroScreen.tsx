import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components/macro';
import Button from '../../../components/Button';
import { textRegularSize } from '../../../components/Typography';
import { nextQuestion } from '../actions';
import * as pqSelectors from '../selectors';

// tslint:disable-next-line: variable-name
const IntroScreenWrapper = styled.div`
  text-align: center;

  ${Button} {
    display: block;
    margin: 2rem auto;
  }
`;

// tslint:disable-next-line: variable-name
const IntroScreenMessage = styled.span`
  ${textRegularSize}
  margin: 4rem 0;
`;

// tslint:disable-next-line: variable-name
const IntroScreen = () => {
  const questionsCount = useSelector(pqSelectors.questionsCount);
  const dispatch = useDispatch();

  return <IntroScreenWrapper>
    <IntroScreenMessage>
      <FormattedMessage id='i18n:practice-questions:popup:intro:message' values={{ questions: questionsCount }}>
        {(msg) => msg}
      </FormattedMessage>
    </IntroScreenMessage>
    <Button
      variant='primary'
      size='large'
      onClick={() => dispatch(nextQuestion())}
      data-analytics-label='Start now'
    >
      <FormattedMessage id='i18n:practice-questions:popup:intro:start'>
        {(msg) => msg}
      </FormattedMessage>
    </Button>
  </IntroScreenWrapper>;
};

export default IntroScreen;
