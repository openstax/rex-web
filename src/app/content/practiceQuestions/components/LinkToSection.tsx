import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useSelector } from 'react-redux';
import styled, { css } from 'styled-components/macro';
import { linkColor } from '../../../components/Typography';
import theme from '../../../theme';
import ContentLink from '../../components/ContentLink';
import * as contentSelectors from '../../selectors';
import { LinkedArchiveTreeSection } from '../../types';

// tslint:disable-next-line: variable-name
export const StyledLink = styled(ContentLink)`
  display: block;
  width: max-content;
  max-width: 100%;
  flex-shrink: 0;
  font-size: 1.4rem;
  color: #929292;
  padding: 2.5rem 0;
  text-decoration: none;

  > span {
    color: ${linkColor};

    &::before {
      content: " ";
    }
  }

  ${theme.breakpoints.mobile(css`
    padding: 1.2rem;
  `)}
`;

interface LinkToSectionProps {
  section: LinkedArchiveTreeSection | null;
}

// tslint:disable-next-line: variable-name
const LinkToSection = ({ section }: LinkToSectionProps) => {
  const book = useSelector(contentSelectors.book);

  if (!section) { return null; }

  return <StyledLink book={book} page={section} target='_blank' data-analytics-label='Go to link' >
    <FormattedMessage id='i18n:practice-questions:popup:read'>
      {(msg) => msg}
    </FormattedMessage>
    <span dangerouslySetInnerHTML={{ __html: section.title }} />
  </StyledLink>;
};

export default LinkToSection;
