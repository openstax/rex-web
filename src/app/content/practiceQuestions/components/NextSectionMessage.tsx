import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components/macro';
import Button from '../../../components/Button';
import { textRegularStyle } from '../../../components/Typography';
import { LinkedArchiveTreeSection } from '../../types';

// tslint:disable-next-line: variable-name
const StyledNextSectionMessage = styled.div`
  max-width: 38rem;
  overflow: initial;

  ${Button} {
    margin: 4rem auto;
  }
`;

// tslint:disable-next-line: variable-name
const StyledMessage = styled.div`
  ${textRegularStyle}
`;

// tslint:disable-next-line: variable-name
export const StyledMessageText = styled.span`
  display: inline;
`;

// tslint:disable-next-line: variable-name
const StyledSectionTitle = styled.span`
  font-weight: 700;
`;

interface NextSectionMessageProps {
  nextSection: LinkedArchiveTreeSection;
  messageKey: string;
  onClick: () => void;
  analyticsLabel?: string;
  className?: string;
}

// tslint:disable-next-line: variable-name
const NextSectionMessage = ({
  nextSection, messageKey, onClick, analyticsLabel, className,
}: NextSectionMessageProps) => (
  <StyledNextSectionMessage className={className}>
    <StyledMessage>
      <StyledMessageText>
        <FormattedMessage id={messageKey}>
          {(msg) => msg}
        </FormattedMessage>
      </StyledMessageText>
      <StyledSectionTitle dangerouslySetInnerHTML={{ __html: nextSection.title }} />
    </StyledMessage>
    <Button
      variant='primary'
      size='large'
      onClick={onClick}
      data-analytics-label={analyticsLabel}
    >
      <FormattedMessage id='i18n:practice-questions:popup:continue'>
        {(msg) => msg}
      </FormattedMessage>
    </Button>
  </StyledNextSectionMessage>
);

export default NextSectionMessage;
