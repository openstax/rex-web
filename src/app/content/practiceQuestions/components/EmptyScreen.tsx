import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useDispatch } from 'react-redux';
import styled from 'styled-components/macro';
import { textRegularStyle } from '../../../components/Typography';
import { LinkedArchiveTreeSection } from '../../types';
import { setSelectedSection } from '../actions';
import NextSectionMessage from './NextSectionMessage';

// tslint:disable-next-line: variable-name
const StyledEmptyScreen = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  margin: 2rem auto;
  flex: 1;
  text-align: center;
  ${textRegularStyle}
`;

// tslint:disable-next-line: variable-name
const StyledText = styled.span`
  margin-bottom: 3rem;
  max-width: 38rem;
  overflow: initial;
`;

interface EmptyScreenProps {
  nextSection: LinkedArchiveTreeSection;
}

// tslint:disable-next-line: variable-name
const EmptyScreen = ({ nextSection }: EmptyScreenProps) => {
  const dispatch = useDispatch();

  return <StyledEmptyScreen>
    <StyledText>
      <FormattedMessage id='i18n:practice-questions:popup:empty:message'>
        {(msg) => msg}
      </FormattedMessage>
    </StyledText>
    <NextSectionMessage
      nextSection={nextSection}
      messageKey='i18n:practice-questions:popup:empty:next-section'
      onClick={() => dispatch(setSelectedSection(nextSection))}
      analyticsLabel='Continue (Empty Screen)'
    />
  </StyledEmptyScreen>;
};

export default EmptyScreen;
