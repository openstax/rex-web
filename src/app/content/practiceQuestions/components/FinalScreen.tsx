import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useDispatch } from 'react-redux';
import styled from 'styled-components/macro';
import { textRegularStyle } from '../../../components/Typography';
import { LinkedArchiveTreeSection } from '../../types';
import { setSelectedSection } from '../actions';
import NextSectionMessage, { StyledMessageText } from './NextSectionMessage';

// tslint:disable-next-line: variable-name
const StyledFinalScreen = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  margin: 2rem auto;
  flex: 1;
  text-align: center;
  ${textRegularStyle}
`;

// tslint:disable-next-line: variable-name
const StyledText = styled.span`
  margin-bottom: 3rem;
  max-width: 38rem;
  overflow: initial;
`;

// tslint:disable-next-line: variable-name
const StyledNextSectionMessage = styled(NextSectionMessage)`
  ${StyledMessageText} {
    display: block;
  }
`;

interface FinalScreenProps {
  nextSection?: LinkedArchiveTreeSection;
}

// tslint:disable-next-line: variable-name
const FinalScreen = ({ nextSection }: FinalScreenProps) => {
  const dispatch = useDispatch();

  return <StyledFinalScreen>
    <StyledText>
      <FormattedMessage id='i18n:practice-questions:popup:final'>
        {(msg) => msg}
      </FormattedMessage>
    </StyledText>
    {
      nextSection &&
      <StyledNextSectionMessage
        nextSection={nextSection}
        messageKey='i18n:practice-questions:popup:final:next-section'
        onClick={() => dispatch(setSelectedSection(nextSection))}
        analyticsLabel='Continue (Final Screen)'
      />
    }
  </StyledFinalScreen>;
};

export default FinalScreen;
