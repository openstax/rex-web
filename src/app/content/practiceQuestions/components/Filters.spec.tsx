import React from 'react';
import renderer from 'react-test-renderer';
import createTestStore from '../../../../test/createTestStore';
import { book } from '../../../../test/mocks/archiveLoader';
import TestContainer from '../../../../test/TestContainer';
import { DropdownToggle } from '../../../components/Dropdown';
import { Store } from '../../../types';
import { receiveBook } from '../../actions';
import { StyledDetails, StyledSectionItem, StyledSummary } from '../../components/popUp/ChapterFilter';
import { LocationFiltersWithChildren } from '../../components/popUp/types';
import { LinkedArchiveTreeSection } from '../../types';
import { setSelectedSection } from '../actions';
import * as selectors from '../selectors';
import Filters from './Filters';

describe('Filters', () => {
  let store: Store;
  let dispatch: jest.SpyInstance;
  let render: () => JSX.Element;
  const mockLocationFilters = new Map([
    ['testbook1-testchapter1-uuid', {
      children: [
        { id: 'testbook1-testpage11-uuid', title: 'section1' },
        { id: 'testbook1-testpage8-uuid', title: 'section2' },
      ],
      section: { id: 'testbook1-testchapter1-uuid', title: 'chapterId' },
    }],
    ['testbook1-testchapter2-uuid', {
      children: [{ id: 'testbook1-testpage3-uuid', title: 'section21' }],
      section: { id: 'testbook1-testchapter2-uuid', title: 'chapterId2' },
    }],
  ]) as LocationFiltersWithChildren;

  beforeEach(() => {
    store = createTestStore();
    dispatch = jest.spyOn(store, 'dispatch');
    render = () => <TestContainer store={store}>
      <Filters />
    </TestContainer>;
  });

  it('ChapterFilter matches snapshot when it is closed', () => {
    const mockFilters = jest.spyOn(selectors, 'practiceQuestionsLocationFilters')
      .mockReturnValue(mockLocationFilters);
    const mockSection = jest.spyOn(selectors, 'selectedSection')
      .mockReturnValue({ id: 'section21', parent: { id: 'chapterId2' } } as LinkedArchiveTreeSection);

    const component = renderer.create(render());

    expect(component).toMatchSnapshot();

    mockFilters.mockReset();
    mockSection.mockReset();
  });

  it('ChapterFilter matches snapshot when open without selected section', () => {
    const mockFilters = jest.spyOn(selectors, 'practiceQuestionsLocationFilters')
      .mockReturnValue(mockLocationFilters);

    const component = renderer.create(render());

    renderer.act(() => {
      const chapterFilterToggle = component.root.findByType(DropdownToggle);
      chapterFilterToggle.props.onClick();
    });

    expect(component).toMatchSnapshot();

    mockFilters.mockReset();
  });

  it('ChapterFilter matches snapshot when open with selected section', () => {
    const mockFilters = jest.spyOn(selectors, 'practiceQuestionsLocationFilters')
      .mockReturnValue(mockLocationFilters);
    const mockSection = jest.spyOn(selectors, 'selectedSection')
      .mockReturnValue({
        id: 'testbook1-testpage3-uuid',
        parent: { id: 'testbook1-testchapter2-uuid' },
      } as LinkedArchiveTreeSection);

    const component = renderer.create(render());

    renderer.act(() => {
      const chapterFilterToggle = component.root.findByType(DropdownToggle);
      chapterFilterToggle.props.onClick();
    });

    expect(component).toMatchSnapshot();

    mockFilters.mockReset();
    mockSection.mockReset();
  });

  it('ChapterFilter with chapters and sections works properly', () => {
    store.dispatch(receiveBook(book));
    const mockFilters = jest.spyOn(selectors, 'practiceQuestionsLocationFilters')
      .mockReturnValue(mockLocationFilters);
    const mockSection = jest.spyOn(selectors, 'selectedSection')
      .mockReturnValue({
        id: 'testbook1-testpage3-uuid',
        parent: { id: 'testbook1-testchapter2-uuid' },
      } as LinkedArchiveTreeSection);

    const component = renderer.create(render());

    renderer.act(() => {
      const chapterFilterToggle = component.root.findByType(DropdownToggle);
      chapterFilterToggle.props.onClick();
    });

    const [details1, details2] = component.root.findAllByType(StyledDetails);
    expect(details1.props.open).toEqual(false);
    expect(details2.props.open).toEqual(true);

    const mockPreventDefault = jest.fn();

    renderer.act(() => {
      const summary = details2.findByType(StyledSummary);
      summary.props.onClick({ preventDefault: mockPreventDefault });
    });

    expect(details1.props.open).toEqual(false);
    expect(details2.props.open).toEqual(false);

    renderer.act(() => {
      const summary = details1.findByType(StyledSummary);
      summary.props.onClick({ preventDefault: mockPreventDefault });
    });

    expect(mockPreventDefault).toHaveBeenCalled();
    expect(details1.props.open).toEqual(true);
    expect(details2.props.open).toEqual(false);

    renderer.act(() => {
      const [section1] = details1.findAllByType(StyledSectionItem);
      section1.props.onClick();
    });

    const expectedSection = { id: 'testbook1-testpage11-uuid', title: 'section1' };

    expect(dispatch).toHaveBeenCalledWith(setSelectedSection(expectedSection as any));

    mockFilters.mockReset();
    mockSection.mockReset();
  });

  it('ChapterFilter does not dispatch setSelectedSection if clicked on already selected section', () => {
    store.dispatch(receiveBook(book));
    const mockFilters = jest.spyOn(selectors, 'practiceQuestionsLocationFilters')
      .mockReturnValue(new Map([
        ['doesnt-matter', {
          children: [
            { id: 'this will not be found in the book', title: 'section1' },
          ],
          section: { id: 'doesnt-matter', title: 'chapterId' },
        }],
      ]) as LocationFiltersWithChildren);
    const mockSection = jest.spyOn(selectors, 'selectedSection')
      .mockReturnValue({
        id: 'this will not be found in the book',
        parent: { id: 'doesnt-matter' },
      } as LinkedArchiveTreeSection);

    dispatch.mockClear();

    const component = renderer.create(render());

    renderer.act(() => {
      const chapterFilterToggle = component.root.findByType(DropdownToggle);
      chapterFilterToggle.props.onClick();
    });

    const [details1] = component.root.findAllByType(StyledDetails);

    renderer.act(() => {
      const summary = details1.findByType(StyledSummary);
      summary.props.onClick({ preventDefault: jest.fn() });
    });

    renderer.act(() => {
      const [section1] = details1.findAllByType(StyledSectionItem);
      section1.props.onClick();
    });

    expect(dispatch).not.toHaveBeenCalled();

    mockFilters.mockReset();
    mockSection.mockReset();
  });
});
