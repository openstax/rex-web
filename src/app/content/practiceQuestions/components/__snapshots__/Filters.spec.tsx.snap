// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters ChapterFilter matches snapshot when it is closed 1`] = `
<div
  className="Filters-sc-1fn81fx-3 gZlQQL"
>
  <div
    className="Filters__FiltersTopBar-sc-1fn81fx-2 fWMshY"
  >
    <div
      className="Dropdown-rmc6yw-5 fBPFfC Dropdown__TabHiddenDropDown-rmc6yw-1 kbMrSv"
    >
      <button
        aria-label="Filter PQ by Chapter & Section"
        className="Dropdown__DropdownToggle-rmc6yw-0 gpiKKF Filters__Toggle-sc-1fn81fx-1 liapzj Button__PlainButton-ayg7nk-2 chgswm"
        data-analytics-label="Filter PQ by Chapter & Section"
        onClick={[Function]}
      >
        <div
          tabIndex={-1}
        >
          Chapter & Section
          <svg
            aria-hidden="true"
            className="Filters__AngleIcon-sc-1fn81fx-0 jYDriK StyledIconBase-sc-bdy9j4 cCvWXa"
            direction="down"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 320 512"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Filters ChapterFilter matches snapshot when open with selected section 1`] = `
<div
  className="Filters-sc-1fn81fx-3 gZlQQL"
>
  <div
    className="Filters__FiltersTopBar-sc-1fn81fx-2 fWMshY"
  >
    <div
      className="Dropdown-rmc6yw-5 fBPFfC Dropdown__TabHiddenDropDown-rmc6yw-1 kbMrSv"
    >
      <button
        aria-label="Filter PQ by Chapter & Section"
        className="Dropdown__DropdownToggle-rmc6yw-0 gpiKKF Filters__Toggle-sc-1fn81fx-1 gsKueT Button__PlainButton-ayg7nk-2 chgswm"
        data-analytics-label="Filter PQ by Chapter & Section"
        onClick={[Function]}
      >
        <div
          tabIndex={-1}
        >
          Chapter & Section
          <svg
            aria-hidden="true"
            className="Filters__AngleIcon-sc-1fn81fx-0 jKNYyQ StyledIconBase-sc-bdy9j4 cCvWXa"
            direction="up"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 320 512"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
      <div
        className="ChapterFilter-tcdlu-7 Filters__StyledChapterFilters-sc-1lvdmlz-0 jynUnN"
        tabIndex={-1}
      >
        <div
          className="ChapterFilter__Row-tcdlu-0 nqIYu"
        >
          <div
            className="ChapterFilter__Column-tcdlu-1 dKPjiE"
          >
            <details
              className="ChapterFilter__StyledDetails-tcdlu-3 inZQMy"
              open={false}
            >
              <summary
                className="ChapterFilter__StyledSummary-tcdlu-4 fDzISg"
                onClick={[Function]}
              >
                <span
                  className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "chapterId",
                    }
                  }
                />
                <svg
                  aria-hidden="true"
                  className="Filters__AngleIcon-sc-1fn81fx-0 jYDriK StyledIconBase-sc-bdy9j4 cCvWXa"
                  direction="down"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 320 512"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                    fill="currentColor"
                  />
                </svg>
              </summary>
              <div
                className="ChapterFilter__StyledChapterFilterItemWrapper-tcdlu-6 jOBTbY"
              >
                <button
                  aria-label="Filter PQ by  section1"
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 hbjjSx"
                  data-analytics-label="Filter PQ by  section1"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section1",
                      }
                    }
                  />
                </button>
                <button
                  aria-label="Filter PQ by  section2"
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 hbjjSx"
                  data-analytics-label="Filter PQ by  section2"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section2",
                      }
                    }
                  />
                </button>
              </div>
            </details>
            <details
              className="ChapterFilter__StyledDetails-tcdlu-3 inZQMy"
              open={true}
            >
              <summary
                className="ChapterFilter__StyledSummary-tcdlu-4 fDzISg"
                onClick={[Function]}
              >
                <span
                  className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "chapterId2",
                    }
                  }
                />
                <svg
                  aria-hidden="true"
                  className="Filters__AngleIcon-sc-1fn81fx-0 jKNYyQ StyledIconBase-sc-bdy9j4 cCvWXa"
                  direction="up"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 320 512"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                    fill="currentColor"
                  />
                </svg>
              </summary>
              <div
                className="ChapterFilter__StyledChapterFilterItemWrapper-tcdlu-6 jOBTbY"
              >
                <button
                  aria-label="Filter PQ by  section21"
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 AJhCb"
                  data-analytics-label="Filter PQ by  section21"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section21",
                      }
                    }
                  />
                </button>
              </div>
            </details>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filters ChapterFilter matches snapshot when open without selected section 1`] = `
<div
  className="Filters-sc-1fn81fx-3 gZlQQL"
>
  <div
    className="Filters__FiltersTopBar-sc-1fn81fx-2 fWMshY"
  >
    <div
      className="Dropdown-rmc6yw-5 fBPFfC Dropdown__TabHiddenDropDown-rmc6yw-1 kbMrSv"
    >
      <button
        aria-label="Filter PQ by Chapter & Section"
        className="Dropdown__DropdownToggle-rmc6yw-0 gpiKKF Filters__Toggle-sc-1fn81fx-1 gsKueT Button__PlainButton-ayg7nk-2 chgswm"
        data-analytics-label="Filter PQ by Chapter & Section"
        onClick={[Function]}
      >
        <div
          tabIndex={-1}
        >
          Chapter & Section
          <svg
            aria-hidden="true"
            className="Filters__AngleIcon-sc-1fn81fx-0 jKNYyQ StyledIconBase-sc-bdy9j4 cCvWXa"
            direction="up"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 320 512"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
      <div
        className="ChapterFilter-tcdlu-7 Filters__StyledChapterFilters-sc-1lvdmlz-0 jynUnN"
        tabIndex={-1}
      >
        <div
          className="ChapterFilter__Row-tcdlu-0 nqIYu"
        >
          <div
            className="ChapterFilter__Column-tcdlu-1 dKPjiE"
          >
            <details
              className="ChapterFilter__StyledDetails-tcdlu-3 inZQMy"
              open={false}
            >
              <summary
                className="ChapterFilter__StyledSummary-tcdlu-4 fDzISg"
                onClick={[Function]}
              >
                <span
                  className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "chapterId",
                    }
                  }
                />
                <svg
                  aria-hidden="true"
                  className="Filters__AngleIcon-sc-1fn81fx-0 jYDriK StyledIconBase-sc-bdy9j4 cCvWXa"
                  direction="down"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 320 512"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                    fill="currentColor"
                  />
                </svg>
              </summary>
              <div
                className="ChapterFilter__StyledChapterFilterItemWrapper-tcdlu-6 jOBTbY"
              >
                <button
                  aria-label="Filter PQ by  section1"
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 hbjjSx"
                  data-analytics-label="Filter PQ by  section1"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section1",
                      }
                    }
                  />
                </button>
                <button
                  aria-label="Filter PQ by  section2"
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 hbjjSx"
                  data-analytics-label="Filter PQ by  section2"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section2",
                      }
                    }
                  />
                </button>
              </div>
            </details>
            <details
              className="ChapterFilter__StyledDetails-tcdlu-3 inZQMy"
              open={false}
            >
              <summary
                className="ChapterFilter__StyledSummary-tcdlu-4 fDzISg"
                onClick={[Function]}
              >
                <span
                  className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "chapterId2",
                    }
                  }
                />
                <svg
                  aria-hidden="true"
                  className="Filters__AngleIcon-sc-1fn81fx-0 jYDriK StyledIconBase-sc-bdy9j4 cCvWXa"
                  direction="down"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 320 512"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                    fill="currentColor"
                  />
                </svg>
              </summary>
              <div
                className="ChapterFilter__StyledChapterFilterItemWrapper-tcdlu-6 jOBTbY"
              >
                <button
                  aria-label="Filter PQ by  section21"
                  className="Button__PlainButton-ayg7nk-2 ChapterFilter__StyledSectionItem-tcdlu-5 hbjjSx"
                  data-analytics-label="Filter PQ by  section21"
                  onClick={[Function]}
                >
                  <span
                    className="ChapterFilter__ChapterTitle-tcdlu-2 fmaoPu"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "section21",
                      }
                    }
                  />
                </button>
              </div>
            </details>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
